source/common/quic/platform/quiche_bug_tracker_impl.h

```diff
@@ -6,11 +6,46 @@
 // consumed or referenced directly by other Envoy code. It serves purely as a
 // porting layer for QUICHE.
 
-#include "source/common/quic/platform/quic_logging_impl.h"
+#include <sstream>
+
+#include "absl/strings/string_view.h"
+
+namespace quic {
+
+class QuicheBugEmitter {
+public:
+  explicit QuicheBugEmitter(absl::string_view condition_str, absl::string_view bug_name)
+      : condition_str_(condition_str), bug_name_(bug_name) {}
+
+  ~QuicheBugEmitter();
+
+  std::ostringstream& stream() { return stream_; }
+
+private:
+  std::ostringstream stream_;
+  const std::string condition_str_;
+  const std::string bug_name_;
+};
+
+class ScopedDisableExitOnQuicheBug {
```

Comment:
nit: Clarify (in comments) it is test only, and only one instance can exists at a time.

Reply:
done. But why only one instance can exists at a time?

Reply:
I don't see comment added?

Suppose two threads, each has a ScopedDisableExitOnQuicheBug instance, called S1 and S2, and S1 and S2's lifetime overlaps like:

T=0:   S1 construct
T=1:                                S2 construct
T=2:   S1 destruct
T=3:                               S2 destruct

The final value for ScopedDisableExitOnQuicheBug will be true.

Reply:
Oh, you mean multiple thread. Add comment for real now.

Reply:
Yes multi thread is the main problem. But that sequence is problematic in single thread too.

Still not seeing comments. LGTM assuming you'll add one:)